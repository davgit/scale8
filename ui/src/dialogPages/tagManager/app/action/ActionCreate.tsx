import { FC } from 'react';
import { ApolloError, useMutation, useQuery } from '@apollo/client';
import CreateActionQuery from '../../../../gql/mutations/CreateActionQuery';
import ActionForm from '../../../../components/organisms/Forms/ActionForm';
import FetchAvailablePlatformActionsQuery from '../../../../gql/queries/FetchAvailablePlatformActionsQuery';
import { FetchAvailablePlatformActions } from '../../../../gql/generated/FetchAvailablePlatformActions';
import { FormWithMappedPlatformValuesResult } from '../../../../hooks/form/useFormWithMappedPlatformValues';
import nameValidator from '../../../../utils/validators/nameValidator';
import { MappedPlatformValues } from '../../../../types/MappedPlatformValuesTypes';
import { ActionCreateInput, DataMapInput } from '../../../../gql/generated/globalTypes';
import { mappedPlatformValuesToDataMapInput } from '../../../../utils/MappedPlatformValuesUtils';
import { CreateActionResult } from '../../../../gql/generated/CreateActionResult';
import { PlatformDataMap } from '../../../../types/DataMapsTypes';
import { AppPlatformRevision, PlatformAction } from '../../../../types/TagRulesTypes';
import { IngestEndpointForEnvironmentSelection } from '../../../../types/IngestEndpointsTypes';
import { DialogPageProps } from '../../../../types/DialogTypes';
import { buildStandardFormInfo } from '../../../../utils/InfoLabelsUtils';
import { ActionFormProps, ActionValues } from '../../../../types/props/forms/ActionFormProps';
import { DialogPreloadForm, DialogPreloadFormProps } from '../../../abstractions/DialogPreloadForm';
import { getActionIcon } from '../../../../utils/TypeIconsUtils';
import { useConfigState } from '../../../../context/AppContext';

export const buildActionCreateInput = (
    actionName: string,
    actionGroupId: string,
    platformActionId: string,
    mappedPlatformValues: MappedPlatformValues,
    comments: string,
): ActionCreateInput => {
    const data_maps: DataMapInput[] = mappedPlatformValuesToDataMapInput(mappedPlatformValues);

    return {
        action_group_id: actionGroupId,
        platform_action_id: platformActionId,
        name: actionName,
        data_maps,
        ...(comments === '' ? {} : { comments }),
    };
};

const ActionCreate: FC<DialogPageProps> = (props: DialogPageProps) => {
    const { consentPurposes, consentVendors } = useConfigState();

    const actionCreateProps: DialogPreloadFormProps<
        FetchAvailablePlatformActions,
        ActionValues,
        ActionFormProps,
        CreateActionResult
    > = {
        loadQuery: useQuery<FetchAvailablePlatformActions>(FetchAvailablePlatformActionsQuery, {
            variables: { id: props.contextId },
        }),
        buildInitialStatePreload: () => ({
            name: '',
            platformActionId: '',
            comments: '',
        }),
        saveQuery: useMutation(CreateActionQuery),
        mapSaveData: (actionValues: ActionValues) => {
            const actionCreateInput = buildActionCreateInput(
                actionValues.name,
                props.id,
                actionValues.platformActionId,
                actionValues.mappedPlatformValues ?? [],
                actionValues.comments,
            );
            return { actionCreateInput };
        },
        buildFormProps: (
            formLoadedData: FetchAvailablePlatformActions,
            formValidationValues: FormWithMappedPlatformValuesResult<ActionValues>,
            gqlError?: ApolloError,
        ): ActionFormProps => ({
            ...formValidationValues,
            gqlError,
            submitText: 'Add Action',
            title: 'Add Action',
            formInfoProps: buildStandardFormInfo('actions', 'Create'),
            handleDialogClose: props.handleDialogClose,
            appPlatformRevisions: formLoadedData.getActionGroupDistribution.revision
                .app_platform_revisions as AppPlatformRevision[],
            environments: formLoadedData.getActionGroupDistribution.revision.app.environments,
            revisions: formLoadedData.getActionGroupDistribution.revision.app.revisions,
            consentPurposes,
            consentVendors,
            ingestEndpoints: formLoadedData.getActionGroupDistribution.revision.app
                .tag_manager_account.org.data_manager_account
                .ingest_endpoints as IngestEndpointForEnvironmentSelection[],
            platformActions:
                formLoadedData.getActionGroupDistribution.revision.app_platform_revisions.map(
                    (_) => ({
                        key: _.platform_revision.platform.id,
                        text: _.platform_revision.platform.name,
                        sub: _.platform_revision.platform_actions.map((a) => {
                            const Icon = getActionIcon(a.icon);

                            return {
                                key: a.id,
                                text: a.name,
                                iconType: a.icon,
                                icon: <Icon />,
                                description: a.description,
                            };
                        }),
                    }),
                ),
        }),
        checkSuccessfullySubmitted: (formMutationData) =>
            formMutationData?.createAction.id !== undefined,
        pageComponent: ActionForm,
        validators: [
            {
                field: 'name',
                validator: nameValidator,
                error: () => 'Action name too short',
            },
        ],
        mappedPlatformValuesProps: {
            buildAvailableModelsWithPlatformDataMaps: (
                formLoadedData: FetchAvailablePlatformActions,
            ) =>
                formLoadedData.getActionGroupDistribution.revision.app_platform_revisions
                    .reduce((accumulator: PlatformAction[], currentValue) => {
                        return [
                            ...accumulator,
                            ...(currentValue.platform_revision
                                .platform_actions as PlatformAction[]),
                        ];
                    }, [])
                    .map((_) => ({
                        id: _.id,
                        platformDataMaps: _.platform_data_maps as PlatformDataMap[],
                    })),
            idValueForModelWithPlatformDataMaps: 'platformActionId',
        },
        ...props,
    };

    return (
        <DialogPreloadForm<
            FetchAvailablePlatformActions,
            ActionValues,
            ActionFormProps,
            CreateActionResult
        >
            {...actionCreateProps}
        />
    );
};

export default ActionCreate;
