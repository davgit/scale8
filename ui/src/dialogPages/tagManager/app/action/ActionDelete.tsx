import { FC } from 'react';
import { DialogPageProps } from '../../../../types/DialogTypes';
import { DialogMutationFunction } from '../../../abstractions/DialogDirectMutation';
import DeleteActionQuery from '../../../../gql/mutations/DeleteActionQuery';
import { DialogDeleteComment } from '../../../abstractions/DialogDeleteComment';

const ActionDelete: FC<DialogPageProps> = (props: DialogPageProps) => {
    return (
        <DialogDeleteComment
            text={`Delete Action ${props.name}?`}
            executeMutationCallback={async (
                mutationFunction: DialogMutationFunction,
                id: string,
                contextId: string,
                ids: string[],
                comments: string,
            ) => {
                await mutationFunction({
                    variables: {
                        actionDeleteInput: {
                            action_id: id,
                            ...(comments === '' ? {} : { comments }),
                        },
                    },
                });
            }}
            mutation={DeleteActionQuery}
            {...props}
        />
    );
};

export default ActionDelete;
