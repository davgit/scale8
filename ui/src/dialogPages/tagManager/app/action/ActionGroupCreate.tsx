import { FC } from 'react';
import ActionGroupForm from '../../../../components/organisms/Forms/ActionGroupForm';
import { FormProps, FormValidationResult } from '../../../../hooks/form/useFormValidation';
import { ApolloError, useMutation } from '@apollo/client';
import CreateActionGroupQuery from '../../../../gql/mutations/CreateActionGroupQuery';
import nameValidator from '../../../../utils/validators/nameValidator';
import { CreateActionGroup } from '../../../../gql/generated/CreateActionGroup';
import { DialogPageProps } from '../../../../types/DialogTypes';
import { buildStandardFormInfo } from '../../../../utils/InfoLabelsUtils';
import { DialogForm, DialogFormProps } from '../../../abstractions/DialogForm';

export type ActionGroupValues = {
    name: string;
    comments: string;
};

export type ActionGroupFormProps = FormProps<ActionGroupValues>;

const ActionGroupCreate: FC<DialogPageProps> = (props: DialogPageProps) => {
    const actionGroupCreateProps: DialogFormProps<
        ActionGroupValues,
        ActionGroupFormProps,
        CreateActionGroup
    > = {
        buildInitialState: () => ({
            name: '',
            comments: '',
        }),
        saveQuery: useMutation(CreateActionGroupQuery),
        mapSaveData: (actionGroupValues: ActionGroupValues) => ({
            actionGroupCreateInput: {
                action_group_distribution_id: props.id,
                name: actionGroupValues.name,
                ...(actionGroupValues.comments === ''
                    ? {}
                    : {
                          comments: actionGroupValues.comments,
                      }),
            },
        }),
        buildFormProps: (
            formValidationValues: FormValidationResult<ActionGroupValues>,
            gqlError?: ApolloError,
        ) => ({
            ...formValidationValues,
            gqlError,
            submitText: 'Create Action Group',
            title: 'Create Action Group',
            formInfoProps: buildStandardFormInfo('actionGroups', 'Create'),
            handleDialogClose: props.handleDialogClose,
        }),
        checkSuccessfullySubmitted: (formMutationData) =>
            formMutationData?.createActionGroup.id !== undefined,
        pageComponent: ActionGroupForm,
        validators: [
            {
                field: 'name',
                validator: nameValidator,
                error: () => 'Action Group name too short',
            },
        ],
        ...props,
    };

    return (
        <DialogForm<ActionGroupValues, ActionGroupFormProps, CreateActionGroup>
            {...actionGroupCreateProps}
        />
    );
};

export default ActionGroupCreate;
