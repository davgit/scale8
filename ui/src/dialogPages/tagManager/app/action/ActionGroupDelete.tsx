import { FC } from 'react';
import DeleteActionGroupQuery from '../../../../gql/mutations/DeleteActionGroupQuery';
import { DeleteActionGroup } from '../../../../gql/generated/DeleteActionGroup';
import { DialogPageProps } from '../../../../types/DialogTypes';
import {
    DialogDeletePreview,
    DialogDeletePreviewProps,
} from '../../../abstractions/DialogDeletePreview';
import { DialogMutationFunction } from '../../../abstractions/DialogDirectMutation';
import { FormMutationData } from '../../../../hooks/form/useFormValidation';
import { modelsDeleteReducer } from '../../../../utils/ArrayUtils';

const ActionGroupDelete: FC<DialogPageProps> = (props: DialogPageProps) => {
    const buildModelsDeletedMap = (data: DeleteActionGroup) =>
        data.deleteActionGroup.reduce(
            modelsDeleteReducer,
            new Map<string, { name: string; id: string }[]>(),
        );

    const findName = (data: FormMutationData | DeleteActionGroup) => {
        const modelsDeletedMap = buildModelsDeletedMap(data as DeleteActionGroup);

        const actionGroupModels = modelsDeletedMap.get('ActionGroup');
        return actionGroupModels ? actionGroupModels[0].name : '';
    };

    const dialogDeletePreviewProps: DialogDeletePreviewProps = {
        mutation: DeleteActionGroupQuery,
        executePreviewMutationCallback: async (
            mutationFunction: DialogMutationFunction,
            id: string,
        ) => {
            await mutationFunction({
                variables: {
                    actionGroupDeleteInput: {
                        action_group_id: id,
                        preview: true,
                    },
                },
            });
        },
        executeMutationCallback: async (
            mutationFunction: DialogMutationFunction,
            id: string,
            contextId: string,
            ids: string[],
            comments: string,
        ) => {
            await mutationFunction({
                variables: {
                    actionGroupDeleteInput: {
                        action_group_id: id,
                        ...(comments === '' ? {} : { comments }),
                    },
                },
            });
        },
        buildText: (data: FormMutationData | DeleteActionGroup) => {
            return `Deleting the action group "${findName(data)}" will also delete:`;
        },
        buildTextNoGroups: (data: FormMutationData | DeleteActionGroup) => {
            return `Delete action group: ${findName(data)}?`;
        },
        buildGroups: (data: FormMutationData | DeleteActionGroup) => {
            const modelsDeletedMap = buildModelsDeletedMap(data as DeleteActionGroup);

            const groups: {
                title: string;
                items: { name: string; id: string }[];
            }[] = [];

            [
                { model: 'Action', label: 'Actions' },
                { model: 'DataMap', label: 'Data Maps' },
                { model: 'RepeatedDataMap', label: 'Repeated Data Maps' },
            ].forEach((_) => {
                const items = modelsDeletedMap.get(_.model);
                if (items !== undefined) {
                    groups.push({
                        title: `${_.label} (${items.length})`,
                        items,
                    });
                }
            });

            return groups;
        },
        ...props,
    };

    return <DialogDeletePreview {...dialogDeletePreviewProps} />;
};

export default ActionGroupDelete;
