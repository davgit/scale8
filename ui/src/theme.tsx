import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
