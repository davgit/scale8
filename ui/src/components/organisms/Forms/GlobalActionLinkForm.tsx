import { FC } from 'react';
import DrawerFormLayout from '../../molecules/DrawerFormLayout';
import ControlledSelect from '../../atoms/ControlledInputs/ControlledSelect';
import { ActionGroupDistributionLinkFormProps } from '../../../dialogPages/tagManager/app/action/ActionGroupDistributionLink';

const GlobalActionLinkForm: FC<ActionGroupDistributionLinkFormProps> = (
    props: ActionGroupDistributionLinkFormProps,
) => {
    return (
        <DrawerFormLayout
            {...props}
            noTitle
            submitDisable={props.isSubmitting || props.availableGlobalActions.length === 0}
        >
            {props.availableGlobalActions.length === 0 ? (
                <small>There are no global action group distributions available.</small>
            ) : (
                <ControlledSelect
                    className="DialogFormField"
                    label="Global Action Group Distribution"
                    name="globalActionGroupDistributionId"
                    values={props.availableGlobalActions}
                    formProps={props}
                    required
                />
            )}
        </DrawerFormLayout>
    );
};

export default GlobalActionLinkForm;
