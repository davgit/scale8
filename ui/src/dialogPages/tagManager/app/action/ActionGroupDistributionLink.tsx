import { FC } from 'react';
import { FormProps, FormValidationResult } from '../../../../hooks/form/useFormValidation';
import { ApolloError, useMutation, useQuery } from '@apollo/client';
import { DialogPageProps } from '../../../../types/DialogTypes';
import { buildStandardFormInfo } from '../../../../utils/InfoLabelsUtils';
import { DialogPreloadForm, DialogPreloadFormProps } from '../../../abstractions/DialogPreloadForm';
import FetchAvailableGlobalActionsQuery from '../../../../gql/queries/FetchAvailableGlobalActionsQuery';
import { FetchAvailableGlobalActions } from '../../../../gql/generated/FetchAvailableGlobalActions';
import GlobalActionLinkForm from '../../../../components/organisms/Forms/GlobalActionLinkForm';
import LinkGlobalActionQuery from '../../../../gql/mutations/LinkGlobalActionQuery';
import { LinkGlobalActionResult } from '../../../../gql/generated/LinkGlobalActionResult';

export type ActionGroupDistributionLinkValues = {
    globalActionGroupDistributionId: string;
};

export type ActionGroupDistributionLinkFormProps = FormProps<ActionGroupDistributionLinkValues> & {
    availableGlobalActions: { key: string; text: string }[];
};

const ActionGroupDistributionLink: FC<DialogPageProps> = (props: DialogPageProps) => {
    const globalActionLinkProps: DialogPreloadFormProps<
        FetchAvailableGlobalActions,
        ActionGroupDistributionLinkValues,
        ActionGroupDistributionLinkFormProps,
        LinkGlobalActionResult
    > = {
        loadQuery: useQuery<FetchAvailableGlobalActions>(FetchAvailableGlobalActionsQuery, {
            variables: { tagId: props.contextId, ruleId: props.id },
        }),
        buildInitialStatePreload: () => ({
            globalActionGroupDistributionId: '',
        }),
        saveQuery: useMutation(LinkGlobalActionQuery),
        mapSaveData: (values: ActionGroupDistributionLinkValues) => ({
            globalActionGroupDistributionLinkInput: {
                rule_id: props.id,
                global_action_group_distribution_id: values.globalActionGroupDistributionId,
            },
        }),
        buildFormProps: (
            formLoadedData: FetchAvailableGlobalActions,
            formValidationValues: FormValidationResult<ActionGroupDistributionLinkValues>,
            gqlError?: ApolloError,
        ) => ({
            ...formValidationValues,
            gqlError,
            submitText: 'Link Global Action Group Distribution',
            title: 'Link Global Action Group Distribution',
            formInfoProps: buildStandardFormInfo('actionGroupDistributions', 'Link'),
            handleDialogClose: props.handleDialogClose,
            availableGlobalActions: formLoadedData.getTag.revision.global_action_group_distributions
                .filter(
                    (_) =>
                        !formLoadedData.getRule.action_groups_distributions
                            .map((d) => d.id)
                            .includes(_.id),
                )
                .map((_) => ({
                    key: _.id,
                    text: _.name,
                })),
        }),
        checkSuccessfullySubmitted: (formMutationData) =>
            formMutationData?.linkGlobalActionGroupDistribution,
        pageComponent: GlobalActionLinkForm,
        validators: [],
        ...props,
    };

    return (
        <DialogPreloadForm<
            FetchAvailableGlobalActions,
            ActionGroupDistributionLinkValues,
            ActionGroupDistributionLinkFormProps,
            LinkGlobalActionResult
        >
            {...globalActionLinkProps}
        />
    );
};

export default ActionGroupDistributionLink;
