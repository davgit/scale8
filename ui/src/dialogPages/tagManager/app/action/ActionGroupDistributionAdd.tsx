import { FC, useState, ChangeEvent } from 'react';
import { DialogPageProps } from '../../../../types/DialogTypes';
import { ActionGroupDistributionCreate } from './ActionGroupDistributionCreate';
import { ActionGroupDistributionLink } from './ActionGroupDistributionLink';
import { InfoButton } from '../../../../components/molecules/InfoButton';
import { buildStandardFormInfo } from '../../../../utils/InfoLabelsUtils';
import AppBar from '@mui/material/AppBar';
import Tabs from '@mui/material/Tabs';
import Tab from '@mui/material/Tab';
import TabsTabPanel from '../../../../components/molecules/TabsTabPanel';
import { Divider } from '@mui/material';
import { MainDrawerTitle } from '../../../../components/molecules/MainDrawerTitle';

const ActionGroupDistributionAdd: FC<DialogPageProps> = (props: DialogPageProps) => {
    const [value, setValue] = useState(0);

    const handleChange = (event: ChangeEvent<unknown>, newValue: number) => {
        setValue(newValue);
    };

    return (
        <>
            <MainDrawerTitle handleDialogClose={props.handleDialogClose}>
                Add Action Group Distribution
                <InfoButton {...buildStandardFormInfo('actionGroupDistributions', 'Add')} />
            </MainDrawerTitle>
            <AppBar position="static" color="default" elevation={0}>
                <Tabs
                    value={value}
                    onChange={handleChange}
                    indicatorColor="primary"
                    textColor="primary"
                    variant="fullWidth"
                >
                    <Tab label="Create Custom" wrapped />
                    <Tab label="Link Global" wrapped />
                </Tabs>
            </AppBar>
            <Divider />
            <TabsTabPanel value={value} index={0}>
                <ActionGroupDistributionCreate
                    {...props}
                    noTitle
                    submitText="Create Custom Action Group Distribution"
                />
            </TabsTabPanel>
            <TabsTabPanel value={value} index={1}>
                <ActionGroupDistributionLink {...props} />
            </TabsTabPanel>
        </>
    );
};

export { ActionGroupDistributionAdd };
