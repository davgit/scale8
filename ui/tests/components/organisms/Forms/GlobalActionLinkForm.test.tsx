import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { FC } from 'react';
import { useMockFormProps } from '../../../componentsMocks/useMockFormProps';
import GlobalActionLinkForm from '../../../../src/components/organisms/Forms/GlobalActionLinkForm';
import { ActionGroupDistributionLinkValues } from '../../../../src/dialogPages/tagManager/app/action/ActionGroupDistributionLink';

const MockForm: FC = () => {
    const formProps = useMockFormProps<ActionGroupDistributionLinkValues>({
        globalActionGroupDistributionId: '',
    });

    return (
        <GlobalActionLinkForm {...formProps} availableGlobalActions={[{ key: 'p1', text: 'p1' }]} />
    );
};

describe('GlobalActionLinkForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
