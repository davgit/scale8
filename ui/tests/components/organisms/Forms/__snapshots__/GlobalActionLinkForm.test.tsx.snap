// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalActionLinkForm form render unchanged 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-s92abg"
  >
    <div
      class="MuiBox-root css-yn9cjk"
    >
      <div
        class="MuiBox-root css-1676va5"
      >
        <form
          class="MuiBox-root css-mjj15b"
        >
          <div
            class="MuiFormControl-root DialogFormField css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
            >
              Global Action Group Distribution
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-foxv06-MuiInputBase-root-MuiInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-globalActionGroupDistributionId"
                class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                id="mui-component-select-globalActionGroupDistributionId"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="globalActionGroupDistributionId"
                required=""
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-yqkyv6-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiBox-root css-wpygli"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-xcbv7o-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
